nanshe_ipython.ipynb

```diff
@@ -1056,8 +1057,11 @@
         "    plot_projs.append(plot_std)\n",
         "\n",
         "\n",
+        "all_tr_dtype_srcs = ColumnDataSource(data=dict(traces_dtype=traces.dtype.type(0)[None]))\n",
```

Comment:
Store a singleton 1D array with a `0` value that is the same type as `traces`. As Bokeh handles conversion of this array to the appropriate type in JavaScript, we can leverage this to cast the decompressed trace data back to the proper type in JavaScript.

Note: While this is a bit of a hack, it is simple, effective, and reliable as it uses Bokeh's standard method of transmission to handle this type. By doing this, we avoid duplicating Bokeh's conversion logic likely resulting in bit rot. Also we avoid tapping into Bokeh's private API likely resulting in breakage.